File: JavaScript/Linked List/206. Reverse Linked List.js
/* 206. Reverse Linked List
Difficulty: Easy
Submission Link: https://leetcode.com/problems/reverse-linked-list/submissions/1391193905/
*/

var reverseList = function(head) {
    let node = null;

    while (head) {
        const temp = head.next;
        head.next = node;
        node = head;
        head = temp;
    }

    return node;    
};

File: C/Linked List/206. Reverse Linked List.c
/* 206. Reverse Linked List
Difficulty: Easy
Submission Link: https://leetcode.com/problems/reverse-linked-list/submissions/1391195351/ */

struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* prev=NULL;
    struct ListNode* temp=head;
    while(temp){
        struct ListNode* next=temp->next;
        temp->next=prev;
        prev=temp;
        temp=next;

    }return prev;
}